jwt-authorizer/templates/README.md

```diff
@@ -0,0 +1,162 @@
+
+6. Select Scopes:
+
+* email
+* profile
+* org.cilogon.userinfo
+
+7. Refresh Token Lifetime - 24 hours
+  - This is not really necessary, we can probably get by without refresh token
+
+Save that information.
+This is your client id and client secret.
+
+### After submission
+
+A separate email is required to CILogonhelp address to apply the client configuration
+from the client `cilogon:/client_id/6ca7b54ac075b65bccb9c885f9ba4a75` to your new
+client.
+
+## Configuring Auth Services
+
+### Prerequisites
+
+- pip, or
+- git, to clone this repo
+- curl, in case you don't have pip or git
+
+#### Source
+Clone this repo, switch to this directory (kube/template).
+
+```
+git clone https://github.com/lsst/jwt_authorizer
+cd jwt_authorizer/kube/template
+```
+
+If you don't have git, you can also get it with curl:
+```
+curl -sSLO https://github.com/lsst/jwt_authorizer/archive/master.tar.gz
+tar xvzf master.tar.gz
+cd jwt-authorizer-master/kube/template
+```
+
+#### Running the init scripts
+init.sh relies on j2cli. By default it will try to use a local version
+then fall back to a version in in docker. If you would like to override that.
+You can override with the J2_BIN for with your own executable. 
+
+You can `pip install j2cli[yaml]` otherwise and it should work too, but it may try to
+install an old version of pyyaml.
+
+If you don't have pip on the system, but at least have python installed, you should be
+able to bootstrap everything, as long as you at least have curl:
+
+```
+curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+python get-pip.py --user
+pip install --user j2cli[yaml]
+```
+
+You will want to modify your PATH too:
+`PATH=$PATH:$HOME/.local/bin:$HOME/bin`
+
+### Run ./init.sh
+This will gather required input and write out YAML files to a directory for 
+your workspace. Those yaml files must be applied.
+
+### Applying the Config and Cleanup
+You can `kubectl apply -R -f .` the config.
+
+The output of this step will, by necessity, contain secrets in the output
+directory. You have some secrets in the clear, you may want to save those 
+somewhere safe (or delete them, if appropriate). The public signing key that
+jwt_authorizer uses to sign tokens is available in a PEM format under
+`public.pem`. *Applications that validate the JWT tokens,
+such as the notebook may need this file*. Other application may rely on the
+JWKS-formatted public key that's deployed to 
+`https://{{ HOSTNAME }}/.well-known/jwks.json`.
+
+
+## Configuring Applications
+
+### Protecting services
+
+Services behind the proxy are configured at the ingress level.
+
+The typical annotation for a browser webapp in kubernetes is configured as follows:
+
+```
+  annotations:
+    kubernetes.io/ingress.class: nginx
+    nginx.ingress.kubernetes.io/auth-request-redirect: $request_uri
+    nginx.ingress.kubernetes.io/auth-response-headers: X-Auth-Request-Token
+    nginx.ingress.kubernetes.io/auth-sign-in: https://{{ HOSTNAME }}/oauth2/sign_in
```

Comment:
here you're saying {{ HOSTNAME }} like it's a template... but it's excluded from helm in the helm ignore?  Also above you say lsst-lsp-instance.example.com?  Is that the same as {{ HOSTNAME }}?

Reply:
This is slightly updated, but I wasn't assuming the end user was using helm with this.

Reply:
This was moved to a helm template in NOTES.txt